src/redux/preview.js

```diff
@@ -0,0 +1,150 @@
+const keyMirror = require('keymirror');
+
+const api = require('../lib/api');
+const log = require('../lib/log');
+
+module.exports.Status = keyMirror({
+    FETCHED: null,
+    NOT_FETCHED: null,
+    FETCHING: null,
+    ERROR: null
+});
+
+module.exports.getInitialState = () => ({
+    status: {
+        project: module.exports.Status.NOT_FETCHED,
+        credit: module.exports.Status.NOT_FETCHED,
+        comments: module.exports.Status.NOT_FETCHED,
+        remixes: module.exports.Status.NOT_FETCHED,
+        studios: module.exports.Status.NOT_FETCHED
+    },
+    projectInfo: {},
+    remixes: [],
+    credit: {},
+    comments: [],
+    studios: []
+});
+
+module.exports.previewReducer = (state, action) => {
+    if (typeof state === 'undefined') {
+        state = module.exports.getInitialState();
+    }
+
+    switch (action.type) {
+    case 'SET_PROJECT_INFO':
+        return Object.assign({}, state, {
+            projectInfo: action.info
+        });
+    case 'SET_REMIXES':
+        return Object.assign({}, state, {
+            remixes: action.items
+        });
+    case 'SET_CREDIT':
+        return Object.assign({}, state, {
+            credit: action.info
+        });
+    case 'SET_COMMENTS':
+        return Object.assign({}, state, {
+            comments: action.items
+        });
```

Comment:
Just a note that this should probably be its own reducer since it'll be reused on other pages.  I think for now it can live here though.